ext/grid.jl

```diff
@@ -4,22 +4,87 @@
 
 function Makie.plot!(plot::Viz{<:Tuple{Grid}})
   grid = plot[:object][]
-  T = typeof(grid)
   Dim = embeddim(grid)
   if Dim == 1
-    vizgrid1D!(T, plot)
+    vizgrid1D!(plot)
   elseif Dim == 2
-    vizgrid2D!(T, plot)
+    vizgrid2D!(plot)
   elseif Dim == 3
-    vizgrid3D!(T, plot)
+    vizgrid3D!(plot)
   end
 end
 
-vizgrid1D!(::Type{<:Grid}, plot) = vizmesh1D!(plot)
+vizgrid1D!(plot) = vizmesh1D!(plot)
 
-vizgrid2D!(::Type{<:Grid}, plot) = vizmesh2D!(plot)
+vizgrid2D!(plot) = vizmesh2D!(plot)
 
-function vizgrid2D!(::Type{<:Union{CartesianGrid,RectilinearGrid}}, plot)
+function vizgrid2D!(plot::Viz{<:Tuple{CartesianGrid}})
+  grid = plot[:object]
+  color = plot[:color]
+  alpha = plot[:alpha]
+  colorscheme = plot[:colorscheme]
+  segmentsize = plot[:segmentsize]
+  showfacets = plot[:showfacets]
+  facetcolor = plot[:facetcolor]
+
+  # process color spec into colorant
+  colorant = Makie.@lift process($color, $colorscheme, $alpha)
+
+  # number of vertices and colors
+  nv = Makie.@lift nvertices($grid)
+  nc = Makie.@lift $colorant isa AbstractVector ? length($colorant) : 1
+
+  # origin, spacing and size of grid
+  or = Makie.@lift coordinates(minimum($grid))
+  sp = Makie.@lift spacing($grid)
+  sz = Makie.@lift size($grid)
+
+  if nc[] == 1
+    # visualize bounding box with a single
+    # color for maximum performance
+    bbox = Makie.@lift boundingbox($grid)
+    viz!(plot, bbox, color=colorant)
+
+    if showfacets[]
+      tup = Makie.@lift xysegments(Meshes.xyz($grid)...)
+      x, y = Makie.@lift($tup[1]), Makie.@lift($tup[2])
+      Makie.lines!(plot, x, y, color=facetcolor, linewidth=segmentsize)
+    end
+  else
+    if nc[] == nv[]
+      # visualize as built-in image with interpolation
+      C = Makie.@lift reshape($colorant, $sz .+ 1)
+      Makie.image!(plot, C, interpolate=true)
+    else
+      # visualize as built-in image without interpolation
+      C = Makie.@lift reshape($colorant, $sz)
+      Makie.image!(plot, C, interpolate=false)
+    end
+
+    if showfacets[]
+      tup = Makie.@lift xysegments(0:$sz[1], 0:$sz[2])
+      x, y = Makie.@lift($tup[1]), Makie.@lift($tup[2])
+      Makie.lines!(plot, x, y, color=facetcolor, linewidth=segmentsize)
+    end
+
+    # adjust spacing and origin
+    spx, spy = sp[]
+    orx, ory = or[]
```

Comment:
do we need to unpack the observables here? Can't we just forward them to Makie.scale! and makie.translate!?

Reply:
Unfortunately no, because `Makie.scale!` and `Makie.translate!` don't accept observables.